core/src/main/java/io/smallrye/openapi/runtime/scanner/spi/AnnotationScanner.java

```diff
@@ -628,7 +631,15 @@ default void processExtensions(final AnnotationScannerContext context, final Met
         }
         for (AnnotationInstance annotation : extensionAnnotations) {
             String name = ExtensionReader.getExtensionName(annotation);
-            operation.addExtension(name, ExtensionReader.readExtensionValue(context, name, annotation));
+            Object value = ExtensionReader.readExtensionValue(context, name, annotation);
+            if (annotation.target() != null && METHOD_PARAMETER.equals(annotation.target().kind())) {
+                final MethodParameterInfo methodParameterInfo = annotation.target().asMethodParameter();
+                final short paramPosition = methodParameterInfo.position();
+                operation.getParameters().get(paramPosition).addExtension(name, value);
```

Comment:
I don't think this approach is going to work outside of the basic scenario. I.e., the position in the operation's parameters may not always be the same as the position in the method parameters. Think of the case where `@BeanParam` is used or where parameters are injected into the resource class itself rather than as method arguments.

Reply:
@MikeEdgar - Any suggestion on how @tkalmar can change the PR so that this will work better ?